packages/generator/src/vdm-types.ts

```diff
@@ -57,11 +57,14 @@ export interface VdmPropertyValueConstraints {
 export interface VdmProperty
   extends VdmPropertyDescriptor,
     VdmPropertyValueConstraints,
-    VdmMappedEdmType {
-  fieldType: string;
+    VdmMappedEdmTypeProperty {
```

Comment:
could you please explain a bit about why we have to introduce `VdmMappedEdmTypeProperty` from this pr?

Reply:
So for the actions I only need the `edmType` and` jsType`  which is the `VdmMappedEdmType`. For the property and complex type I needed in addition the `fieldType` which is then the `VdmMappedEdmTypeProperty` . I could also just add use for example the `VdmMappedEdmType` add the `fieldType` also there and for the action/function case this is not used. Do you yhink this is cleaner?

Reply:
After some thought I will remove the extra type and just use the same one for the two.